Travis: Build with Qt4 error

Extract from https://github.com/kvirc/KVIrc/pull/1828

Build from WANT_KDE=YES - error

```
[ 51%] Building CXX object src/kvirc/CMakeFiles/kvirc.dir/ui/KviInputHistory.cpp.o
/home/travis/build/kvirc/KVIrc/src/kvirc/ui/KviInput.cpp: In member function ‘virtual void KviInput::keyPressEvent(QKeyEvent*)’:
/home/travis/build/kvirc/KVIrc/src/kvirc/ui/KviInput.cpp:253:27: warning: unused variable ‘pYesButton’ [-Wunused-variable]
          QAbstractButton *pYesButton = pMsgBox.addButton(__tr2qs("Yes"), QMessageBox::YesRole);
                           ^
[ 51%] Building CXX object src/kvirc/CMakeFiles/kvirc.dir/ui/KviIpEditor.cpp.o
/home/travis/build/kvirc/KVIrc/src/kvirc/ui/KviInputEditor.cpp: In member function ‘void KviInputEditor::rebuildTextBlocks()’:
/home/travis/build/kvirc/KVIrc/src/kvirc/ui/KviInputEditor.cpp:637:11: warning: unused variable ‘iNextChar’ [-Wunused-variable]
       int iNextChar = KviControlCodes::getUnicodeColorBytes(spb->szText,p - spb->szText.unicode(),&uFore,&uBack);
           ^
/home/travis/build/kvirc/KVIrc/src/kvirc/ui/KviInputEditor.cpp: In member function ‘virtual void KviInputEditor::drawContents(QPainter*)’:
/home/travis/build/kvirc/KVIrc/src/kvirc/ui/KviInputEditor.cpp:844:6: warning: unused variable ‘iCurChar’ [-Wunused-variable]
  int iCurChar = 0;
      ^
/home/travis/build/kvirc/KVIrc/src/kvirc/ui/KviInputEditor.cpp: In member function ‘virtual QVariant KviInputEditor::inputMethodQuery(Qt::InputMethodQuery) const’:
/home/travis/build/kvirc/KVIrc/src/kvirc/ui/KviInputEditor.cpp:1778:8: error: ‘ImEnabled’ is not a member of ‘Qt’
   case Qt::ImEnabled:
        ^
/home/travis/build/kvirc/KVIrc/src/kvirc/ui/KviInputEditor.cpp:1781:8: error: ‘ImHints’ is not a member of ‘Qt’
   case Qt::ImHints:
        ^
[ 51%] Building CXX object src/kvirc/CMakeFiles/kvirc.dir/ui/KviIrcToolBar.cpp.o
make[2]: *** [src/kvirc/CMakeFiles/kvirc.dir/ui/KviInputEditor.cpp.o] Error 1
```

@ctrlaltca said

https://github.com/kvirc/KVIrc/commit/cad079d188f25c90b1c9a6e3add73c2c782febbb#diff-c3aa4a4955c05f5ddf447e38c25b5fd6R1778 is qt5-only, needs a proper ```#ifdef```

Also, the travis script is broken; the build errored out, but the script is reporting success: https://travis-ci.org/kvirc/KVIrc/jobs/101120874


@DarthGandalf @Dessa  mind having a look at travis bits pls? :wink: 

@wodim any ideas about the #ifdef or shall we pester pragma?

--- a/src/kvirc/ui/KviInputEditor.cpp
+++ b/src/kvirc/ui/KviInputEditor.cpp
@@ -1771,6 +1771,7 @@ void KviInputEditor::internalCursorLeft(bool bShift)
 	
 }
 
+#if (QT_VERSION >= 0x050000)
 QVariant KviInputEditor::inputMethodQuery(Qt::InputMethodQuery query) const
 {
 	switch(query)
@@ -1788,6 +1789,7 @@ QVariant KviInputEditor::inputMethodQuery(Qt::InputMethodQuery query) const
 
 	return QWidget::inputMethodQuery(query);
 }
+#endif
 
 
 void KviInputEditor::inputMethodEvent(QInputMethodEvent * e)

--- a/src/kvirc/ui/KviInputEditor.h
+++ b/src/kvirc/ui/KviInputEditor.h
@@ -926,7 +926,9 @@ protected:
 	virtual void dragEnterEvent(QDragEnterEvent * e);
 	virtual void dropEvent(QDropEvent * e);
 	virtual void inputMethodEvent(QInputMethodEvent * e) ;
+#if (QT_VERSION >= 0x050000)
 	virtual QVariant inputMethodQuery(Qt::InputMethodQuery query) const;
+#endif
 	virtual void paintEvent(QPaintEvent * e);
 	bool checkWordSpelling(const QString &szWord);
 	void splitTextIntoSpellCheckerBlocks(const QString &szText,KviPointerList<KviInputEditorSpellCheckerBlock> &lBuffer);

